build.gradle

```diff
@@ -260,6 +263,7 @@ ext.lib = [
         slf4j_api: "org.slf4j:slf4j-api:1.7.21",
         logback: "ch.qos.logback:logback-classic:1.1.7",
         zip4j: "net.lingala.zip4j:zip4j:1.3.2",
+        gradle_core:"org.gradle:gradle-core:6.1.1",
```

Comment:
I have one concern in bringing this dependency here. Although this library doesn't seem to be very heavy, it seems to bring many transitive dependencies that are really not necessary for Butterfly. That would increase the memory footprint and tool binaries considerably.

Please, take a look at the transitive dependencies that would be brought by this dependency, and also think about alternative solutions.

Reply:
I agree about the transitive dependency. As Butterfly doesn't have knowledge about the previous versions, like it has about the next version through `UpgradeStep`, we need to compare two versions to find if the `targetVersion` is older and provide specific exception message.

I can implement the function. In that case, what versioning style should I cover? 
Like-
```
X.Y.Z
X.Y.Z-alpha
X.Y.Z-beta
X.Y.Z-snapshot
```

Also, while I was googling I found [this](https://github.com/G00fY2/version-compare) library which seems to be very light. 

What do you suggest? Please let me know if I have missed something in the code base.

Reply:
Let's try this new library you found please. Thanks.

Reply:
Made necessary changes.